apis/chef/src/main/java/org/jclouds/chef/domain/Role.java

```diff
@@ -85,8 +92,39 @@ public Builder runList(Iterable<String> runList) {
          return this;
       }
 
+      public Builder envRunList(Map<String, List<String>> envRunList) {
+         this.envRunList.putAll(checkNotNull(envRunList, "envRunList"));
+         return this;
+      }
+
+      public Builder envRunList(String name, List<String> runList) {
+         this.envRunList.put(checkNotNull(name, "name"), checkNotNull(runList, "runList"));
+         return this;
+      }
+
+      public Builder envRunListElement(String name, String value) {
+         checkNotNull(name, "name");
+         checkNotNull(value, "value");
+         List<String> runList = this.envRunList.get(name);
+         if (runList == null) {
+            runList = new ArrayList<String>();
+            this.envRunList.put(name, runList);
+         }
+         runList.add(value);
+         return this;
+      }
+
       public Role build() {
-         return new Role(name, description, defaultAttributes.build(), runList.build(), overrideAttributes.build());
+         // Assemble an immutable envRunList where each entry is an immutable list of entries.
+         ImmutableMap.Builder<String, List<String>> immutableEnvRunList = ImmutableMap.builder();
+         for (Entry<String, List<String>> e : envRunList.entrySet()) {
+            ImmutableList.Builder<String> value = ImmutableList.builder();
+            value.addAll(e.getValue());
+            immutableEnvRunList.put(e.getKey(), value.build());
```

Comment:
No need for a builder and a list here. Something like the following is more concise and easier to read?
```java
immutableEnvRunList.put(e.getKey(), ImmutableList.copyOf(e.getValue()));
```
Or use the more functional approach:
```java
immutableEnv = Maps.transformValues(envRunList, new Function<List<String>, List<String>>() {
   @Override public List<String> apply(List<String> input) {
      return ImmutableList.copyOf(input);
   }
});
```